File: cache-bust.js
#!/usr/bin/env node

const fs = require('fs');
const path = require('path');

// Generate a cache-busting version (timestamp)
const version = Date.now();

// Files to update (your HTML files)
const htmlFiles = [
  'index.html',
  // Add other HTML files here
];

// Process each HTML file
htmlFiles.forEach(file => {
  const filePath = path.join(__dirname, file);
  
  if (!fs.existsSync(filePath)) {
    console.log(`⚠️  Skipping ${file} (not found)`);
    return;
  }
  
  let content = fs.readFileSync(filePath, 'utf8');
  
  // Update CSS links
  content = content.replace(
    /(<link[^>]*href=["'])([^"'?]+\.css)(\?v=\d+)?(["'][^>]*>)/gi,
    `$1$2?v=${version}$4`
  );
  
  // Update JS script sources
  content = content.replace(
    /(<script[^>]*src=["'])([^"'?]+\.js)(\?v=\d+)?(["'][^>]*>)/gi,
    `$1$2?v=${version}$4`
  );
  
  // Update image sources (optional)
  content = content.replace(
    /(<img[^>]*src=["'])([^"'?]+\.(png|jpg|jpeg|gif|svg|webp))(\?v=\d+)?(["'][^>]*>)/gi,
    `$1$2?v=${version}$5`
  );
  
  fs.writeFileSync(filePath, content);
  console.log(`✅ Updated ${file} with version ${version}`);
});

console.log('\n🎉 Cache busting complete!');

File: cache-bust.py
#!/usr/bin/env python3

import os
import re
import time
from pathlib import Path

# Generate cache-busting version (timestamp)
version = int(time.time())

# Find all HTML files in the current directory and subdirectories
html_files = list(Path('.').rglob('*.html'))

if not html_files:
    print("⚠️  No HTML files found in directory")
    exit(1)

print(f"Found {len(html_files)} HTML file(s)")
print(f"Using version: {version}\n")

# Process each HTML file
for file_path in html_files:
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        original_content = content
        
        # Update CSS links
        content = re.sub(
            r'(<link[^>]*href=["\'])([^"\'?]+\.css)(\?v=\d+)?(["\'][^>]*>)',
            rf'\1\2?v={version}\4',
            content,
            flags=re.IGNORECASE
        )
        
        # Update JS script sources
        content = re.sub(
            r'(<script[^>]*src=["\'])([^"\'?]+\.js)(\?v=\d+)?(["\'][^>]*>)',
            rf'\1\2?v={version}\4',
            content,
            flags=re.IGNORECASE
        )
        
        # Update image sources (optional)
        content = re.sub(
            r'(<img[^>]*src=["\'])([^"\'?]+\.(png|jpg|jpeg|gif|svg|webp))(\?v=\d+)?(["\'][^>]*>)',
            rf'\1\2?v={version}\5',
            content,
            flags=re.IGNORECASE
        )
        
        # Only write if content changed
        if content != original_content:
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(content)
            print(f"✅ Updated {file_path}")
        else:
            print(f"⏭️  No changes needed for {file_path}")
            
    except Exception as e:
        print(f"❌ Error processing {file_path}: {e}")

print(f"\n🎉 Cache busting complete! Version: {version}")